src/js/views/Project/ViewScoreDetails.jsx

```diff
@@ -0,0 +1,153 @@
+    (value.data_type === 'boolean' && value.value !== true)
+      ? 'bg-red-200'
+      : value.data_type === 'boolean' && value.value === true
+      ? 'bg-green-200'
+      : ''
+  return (
+    <div className="grid grid-cols-2 text-sm ml-2">
+      <div className="font-medium">Weight</div>
+      <div>
+        {value.weight} / {totalWeight}
+      </div>
+      <div className={`${valueClassName} font-medium`}>Value</div>
+      <div className={valueClassName}>
+        <FactValue value={value} />
+      </div>
+      <div className="font-medium">Score</div>
+      <div>{value.score}</div>
+    </div>
+  )
+}
+
+const EnumValues = ({ enums }) => {
+  return (
+    <div className="grid grid-cols-2 text-sm mt-2 ml-2">
+      {enums.map(({ score, selected, value }) => (
+        <Fragment key={value}>
+          <div className={scoreClass(score, selected)}>{value}</div>
+          <div className={scoreClass(score, selected)}>{score}</div>
+        </Fragment>
+      ))}
+    </div>
+  )
+}
+
+const RangeValues = ({ ranges }) => {
+  return (
+    <div className="grid grid-cols-2 text-sm mt-2 ml-2">
+      {ranges.map(({ value, score, selected }) => (
+        <Fragment key={`${value[0]}-${value[1]}`}>
+          <div className={scoreClass(score, selected)}>
+            {value[0]}-{value[1]}
+          </div>
+          <div className={scoreClass(score, selected)}>{score}</div>
+        </Fragment>
+      ))}
+    </div>
+  )
+}
+
+const FactValue = ({ value }) => {
+  const { t } = useTranslation()
+  if (value.value === null || value.value === undefined) {
+    return <div>{t('common.notSet')}</div>
+  }
+  if (value.data_type === 'boolean') {
+    if (value.value === true) {
+      return <Icon className="text-green-600" icon="fas check" />
+    }
+    return <Icon className="text-red-600" icon="fas times-circle" />
+  }
+  return <div>{value.value}</div>
+}
+
+const FactDescription = ({ value, totalWeight }) => {
+  switch (value.fact_type) {
+    case 'enum':
+      return (
+        <Definition term={value.name}>
+          <FactHeading value={value} totalWeight={totalWeight} />
+          <EnumValues enums={value.enums} />
+        </Definition>
+      )
+    case 'range':
+      return (
+        <Definition term={value.name}>
+          <FactHeading value={value} totalWeight={totalWeight} />
+          <RangeValues ranges={value.ranges} />
+        </Definition>
+      )
+    default:
+      return (
+        <Definition term={value.name}>
+          <FactHeading value={value} totalWeight={totalWeight} />
+        </Definition>
+      )
+  }
+}
+
+export function ViewScoreDetails({ project }) {
+  const [globalState] = useContext(Context)
+  const [fetching, setFetching] = useState(false)
+  const [errorMessage, setErrorMessage] = useState(null)
+  const [facts, setFacts] = useState([])
+  const [totalWeight, setTotalWeight] = useState(0)
+  useEffect(() => {
+    setFetching(true)
+    setFacts([])
+    setTotalWeight(0)
```

Comment:
You can probably initialize `fetching` to `true` and then remove these 3 lines since that's the starting state anyway and this hook runs on initial render.

Reply:
good point ... _going to have to watch out since this is somewhat patternistic_